### ISSUE TITLE
Feature: support long options

### ISSUE THREAD

**Issue by joshrabinowitz** [sobolevn/git-secret#330]
It would be nice if git-secret supported short and long command-line options, such as --verbose instead of -v and --preserve instead of -P.

Support for long-form command line options also makes code and examples using git-secret more self-documenting. 

Implementation details aside, I think this is a feature we should add. Any thoughts from users or developers are welcome.

**Comment by joshrabinowitz** [comment]
See also #395

**Comment by sobolevn** [comment]
I agree, since there are a lot of options it may be more friendly to have long alternatives.

**Comment by joshrabinowitz** [comment]
It would be great for someone to make a first pass at defining a long option version of each option we support now. Hint hint to anyone : ]
